x-pack/plugins/elastic_assistant/server/lib/langchain/elasticsearch_store/elasticsearch_store.ts

```diff
@@ -290,6 +296,28 @@ export class ElasticsearchStore extends VectorStore {
     }
   };
 
+  /**
+   * Checks if a kbResource document exists in the Knowledge Base
+   *
+   * @param kbResource
+   */
+  kbResourceExists = async (kbResource: string): Promise<boolean> => {
```

Comment:
If you're okay with it, for these 'legacy' checks, let's move them to the `kbDataClient` and explicitly name them `legacy`. It would be nice to delete this file as soon as possible as I found it pretty difficult to reason between all the branches in esStore between the two different KB mapping versions.